plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt

```diff
@@ -825,27 +845,43 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase {
     rotationDegreesSinceLast: Float
   ): Boolean {
     // in order not to mess up initial anchor values
-    if (immediateEaseInProcess) {
+    if (!internalSettings.enablePinchToZoomWhenRotating && immediateEaseInProcess) {
       return true
     }
     // Calculate map bearing value
     val currentBearing = mapCameraManagerDelegate.cameraState.bearing
     rotateCachedAnchor = cameraAnimationsPlugin.anchor
     val bearing = currentBearing + rotationDegreesSinceLast
-
     // Rotate the map
-    val focalPoint = getRotateFocalPoint(detector)
-    easeToImmediately(
-      CameraOptions.Builder()
-        .anchor(focalPoint)
-        .bearing(bearing)
-        .build(),
-      actionAfter = {
-        cameraAnimationsPlugin.anchor = rotateCachedAnchor
-        notifyOnRotateListeners(detector)
-      }
-    )
-
+    if (internalSettings.enablePinchToZoomWhenRotating) {
+      val bearingAnimator =
+        cameraAnimationsPlugin.createBearingAnimator(
+          cameraAnimatorOptions(bearing) {
+            owner(MapAnimationOwnerRegistry.GESTURES)
+          }
+        ) {
+          duration = 0
+        }
+      bearingAnimator.addListener(
+        onEnd = {
+          cameraAnimationsPlugin.anchor = rotateCachedAnchor
+          notifyOnRotateListeners(detector)
```

Comment:
do you have tests for such case?

Reply:
It seems we don't have tests for animators now. @kiryldz please fix me if I'm wrong.

Reply:
guess this could be covered with unit test to verify that `notifyOnRotateListeners` got called. 
Gesture instrument tests are also present in `app/src/androidTest/java/com/mapbox/maps/testapp/gestures/GestureActivityTest.kt` so feel free to add tests you like 😉